packages/flutter_tools/lib/src/commands/build_bundle.dart

```diff
@@ -36,8 +39,12 @@ class BuildBundleCommand extends BuildSubCommand {
           'linux-x64',
           'windows-x64',
         ],
+        help: '??? This argument is undocumented! ???',
       )
-      ..addOption('asset-dir', defaultsTo: getAssetBuildDirectory());
+      ..addOption('asset-dir',
+        defaultsTo: getAssetBuildDirectory(),
+        help: '??? This argument is undocumented! ???',
```

Comment:
This is an odd one. By default this is `build/flutter_assets` - it controls the build subfolder where the tool places the kernel_blob/native snapshot and the assets/ subfolder, for example: build/flutter_assets/kernel_blob.bin.  its mostly useful if you have your own tooling built on flutter.

Ideally it would have been called something like "output-dir", but I think its been around for a while.